mod/wikirate_source/set/type/source/source.rb

```diff
@@ -63,11 +69,49 @@ def indirect_contributor_search_args
         errors.add :link, "exists already. <a href='/#{duplicated_name}'>Visit the source.</a>"   
       end
     end
+    if errors.empty? and file_link url
+      download_file_and_add_to_plus_file url
+    end   
     parse_source_page url if Card::Env.params[:sourcebox] == 'true'
   end
   
```

Comment:
I really want to start getting away from these long methods (including events and views).  We want our methods to average under 5 lines.  It seems like yours are averaging about 30.   This one is 36.

Shorter methods are easier to test, debug, maintain etc.  They also involve a lot of built-in documentation, assuming the names are well chosen.